ddl/table.go

```diff
@@ -156,11 +172,40 @@ func onCreateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 	return ver, errors.Trace(err)
 }
 
+func createTableWithForeignKeys(d *ddlCtx, t *meta.Meta, job *model.Job, tbInfo *model.TableInfo, fkCheck bool) (ver int64, err error) {
+	switch job.SchemaState {
```

Comment:
```suggestion
	switch tbInfo.State {
```

The job's schema state should be only used in debug or test.